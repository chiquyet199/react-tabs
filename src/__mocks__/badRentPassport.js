export default {
  userId: "67e4b887-466e-4abe-8d12-bc87204e5df8",
  status: "SUBMITTED",
  aboutInfo: {
    email: "jonathan@gmail.com",
    firstName: "Jonathan",
    lastName: "Beaumont",
    middleName: null,
    gender: "MALE",
    dayOfBirth: 1,
    monthOfBirth: 1,
    yearOfBirth: 1970,
    maritalStatus: "MARRIED",
    phoneNumber: "01123456789",
    nationality: "UK",
    isSmoker: true,
    hasPets: true,
    hasChildren: true,
    status: "GREEN",
    messages: [],
  },
  annualIncomeInfo: {
    salary: 100000,
    pension: 0,
    interest: 0,
    benefits: 0,
    rentals: 0,
    other: 0,
    grossAnnualIncome: 100000,
    sourcesOfIncome: ["Employment"],
    status: "GREEN",
    messages: [],
  },
  financialInfo: {
    bankrupt: false,
    creditScore: 790,
    depositFree: true,
    maximumEligible: 1800.266,
    activeCCJs: 1450,
    activeCCJYear: 2016,
    activeCCJMonth: 1,
    activeCCJValue: 0,
    previousCCJs: 0,
    previousCCJsYear: 0,
    previousCCJsMonth: 0,
    activeInsolvencies: 3250,
    activeInsolvenciesYear: 2016,
    activeInsolvenciesMonth: 11,
    previousInsolvencies: false,
    previousInsolvenciesYear: 0,
    previousInsolvenciesMonth: 0,
    status: "YELLOW",
    messages: [
      "credit.insolvency.failed",
      "credit.countyCourtJudgments.failed",
    ],
  },
  legalInfo: {
    pep: true,
    mortality: "failed",
    legalWarnings: true,
    sanctions: true,
    status: "RED",
    messages: [
      "watchList.politicallyExposedPerson.failed",
      "watchList.legalAndRegulatoryWarnings.failed",
      "watchList.sanction.failed",
      "identity.mortality.failed",
    ],
  },
  workInfo: [
    {
      workStatus: "EMPLOYED",
      companyName: "companyName",
      jobTitle: "jobTitle",
      income: 100000,
      employmentTime: "FULL_TIME",
      employmentType: "CONTRACT",
      workedSinceMonth: 12,
      workedSinceYear: 2017,
      studentType: "POST_GRADUATE_STUDENT",
      reference: {
        contactFirstName: "Silvester",
        contactLastName: "Stallone",
        contactEmailAddress: "rambo666@gmail.com",
        referredAt: "12-12-2016",
      },
      status: "YELLOW",
      messages: ["requestReference"],
    },
  ],
  residenceInfo: [
    {
      line1: "Addresss line 1",
      town: "Town",
      state: "State",
      postCode: "UL11223",
      countryCode: "UK",
      residenceType: "RENTING_FROM_AGENT",
      livedSinceMonth: 1,
      livedSinceYear: 2016,
      livedToMonth: null,
      livedToYear: null,
      reference: {
        contactFirstName: "Jonathan",
        contactLastName: "Beaumont",
        contactEmailAddress: "jonathan@gmail.com",
        referredAt: "01-01-2018",
      },
      status: "GREEN",
      messages: [],
    },
  ],
};
