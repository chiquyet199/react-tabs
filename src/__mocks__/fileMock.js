export function MockFile() {}

MockFile.prototype.create = (
  name = "mock.jpg",
  size = 1024,
  mimeType = "image/png",
) => {
  function range(count) {
    let output = "";
    for (let i = 0; i < count; i += 1) {
      output += "a";
    }

    return output;
  }

  const blob = new Blob([range(size)], {type: mimeType});
  blob.lastModifiedDate = new Date();
  blob.name = name;

  return blob;
};
