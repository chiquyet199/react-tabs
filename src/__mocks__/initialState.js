export const initialState = {
  locale: {
    currentLocale: "en-US",
    translations: {
      common: {
        postcode_lookup: {
          postcode: "Postcode",
          find_address: "Find address",
          address_not_listed: "Address not listed",
        },
      },
      home__welcome: "Welcome to RNW starter app",
      home__loggedin: "You are now logged in",
      login__email__label: "Email address",
      login__password__label: "Password",
      login__email__placeholder: "example@example.com",
      login__password__placeholder: "******",
      login__continue__with__facebook: "Continue with Facebook",
      login: "Sign in",
      create__account: "Create account",
      login__forgot__password: "Forgot password",
      logout: "Logout",
      back: "Back",
      submit: "Submit",
      rent_passport: "Rent Passport",
      deposit_free: "Deposit Free",
      home_management: "Home Management",
      authorize: "Authorize",
      terms_header: "Canopy terms & conditions",
      terms:
        "1. INTRODUCTION\n\nBy signing up for an account with Canopy, you are agreeing to be bound by the following terms and conditions (“Terms of Service”). These Terms of Service are to be read in conjunction with:\n\nour Privacy Policy, which sets out the terms on which we process any personal data we collect from you, or that you or others provide to us; and\n\nour Cookies Policy, which sets out information about the cookies we use.\n\nThese Terms of Service are important. Please read them carefully as they contain details of our statutory and regulatory responsibilities to you, and your contractual obligations to us. They also set out the basis on which we provide, and you are permitted to use, the various services made available on the website www.findyourcanopy.com and the mobile application known as ‘Canopy’ (together, the “Platform”), including the following:",
      verify__email__title__text: "Verify your email address",
      verify__email__text:
        "We need to know you are you before creating a RentPassport. Please click the link in the email we sent you.",
      open__email__client: "Open my email client",
      resend__verification__email: "Resend verification email",
      not__received__an__email: "Not received an email?",
      forgot_password: "Forgot your password?",
      reset_password: "Change your password",
      tenant_reference_submitted: "Reference submitted",
      tenant_reference_thanks_for_time:
        "Thanks for taking the time to submit a tenant reference for",
      tenant_reference_did_you_know:
        "Did you know that Canopy offers the following property management tools to tenants, landlords and agents:",
      tenant_reference_rent_passport_text:
        "Full tenant referencing and property management including deposits and payments",
      tenant_reference_deposit_free_text:
        "Qualified tenants can pay a premium to cover their rental security deposit",
      tenant_reference_home_management_text:
        "With a valid UK Open Banking account, tenants are able to have rent payments contribute to their Experian credit report",
      find_out_more: "Find out more",
      tenant_reference_title: "Tenant reference request",
      upload_tenant_reference_document: "Upload tenant reference document",
      upload_tenant_reference_document_description:
        "If you have a manual form you usually use, please upload it here.",
      upload: "Upload",
      submit_reference: "Submit reference",
      tenant_reference_request: "Tenant reference request",
      complete_reference_form_description:
        " tenant of yours, has authorised us to request a reference from you.\n" +
        "As you know a landlord reference is significant information when determining eligibility of a prospective tenant, so please consider the questions below carefully and feel free to make additional comments.",
      upload_reference_document: "Upload reference document",
      complete_reference_form: "Complete reference form",
      complete_reference_form_a: ", a ",
      truelayer: {
        truelayer_auth_title: "Truelayer Authorize",
      },
      employer_reference_form_description:
        " has advised us that you are their employer and has authorised us to request a reference from you.\n" +
        "As you know an employment reference is significant information when determining eligibility of a prospective tenant, so please consider the questions below carefully and feel free to make additional comments.",
    },
  },
  auth: {
    user: null,
    isLoggedIn: false,
  },
  dimensions: {
    fontScale: 1,
    height: 901,
    scale: 1,
    width: 1920,
    size: "L",
  },
  inputField: {
    showPassword: true,
  },
  form: {},
  partialForm: {
    values: {},
  },
  progress: 0,
  notification: {
    type: null,
    message: "",
  },
  user: {
    profile: {
      firstName: "FirstName",
      lastName: "LastName",
    },
  },
  router: {
    location: {
      pathname: "/tenant-reference-request/complete",
      search: "",
      hash: "",
    },
    action: "POP",
  },
  lookups: {
    addresses: [],
  },
  topbar: {
    haveTopbar: true,
    topbarBtn: {
      haveTopbarButton: false,
      topbarButtonText: "",
      btnType: "transparent_green",
      topbarButtonIcon: undefined,
      hideBackButton: false,
    },
    topbarCenter: {
      topbarCenterText: "Center",
    },
    topbarLeft: {
      topbarLeftText: "Back",
    },
  },
  images: {
    accepted: [],
  },
  rentPassport: {
    aboutYou: {
      email: "jonathan@gmail.com",
      firstName: "Jonathan",
      lastName: "Beaumont",
      middleName: null,
      gender: "MALE",
      dayOfBirth: 20,
      monthOfBirth: 11,
      yearOfBirth: 1970,
      maritalStatus: "MARRIED",
      phoneNumber: "01123456789",
      nationality: "UK",
      isSmoker: true,
      hasPets: true,
      hasChildren: true,
      status: "GREEN",
      messages: [],
    },
    financial: {
      salary: 100000,
      pension: 0,
      interest: 0,
      benefits: 0,
      rentals: 0,
      other: 0,
      grossAnnualIncome: 100000,
      sourcesOfIncome: ["Employment"],
      status: "YELLOW",
      messages: [],
      bankrupt: false,
      creditScore: 930,
      depositFree: true,
      maximumEligible: 1800.266,
      activeCCJs: 0,
      activeCCJYear: 0,
      activeCCJMonth: 0,
      activeCCJValue: 0,
      previousCCJs: 0,
      previousCCJsYear: 0,
      previousCCJsMonth: 0,
      activeInsolvencies: 0,
      activeInsolvenciesYear: 0,
      activeInsolvenciesMonth: 0,
      previousInsolvencies: false,
      previousInsolvenciesYear: 0,
      previousInsolvenciesMonth: 0,
    },
    rentPassportLegal: {
      pep: false,
      mortality: "passed",
      legalWarnings: false,
      sanctions: false,
      status: "GREEN",
      messages: [],
    },
    whatYouDo: {
      workInfo: [
        {
          workStatus: "EMPLOYED",
          companyName: "companyName",
          jobTitle: "jobTitle",
          income: 100000,
          employmentTime: "FULL_TIME",
          employmentType: "CONTRACT",
          workedSinceMonth: 12,
          workedSinceYear: 2017,
          studentType: "POST_GRADUATE_STUDENT",
          reference: {
            contactFirstName: "Silvester",
            contactLastName: "Stallone",
            contactEmailAddress: "rambo666@gmail.com",
            referredAt: "12-12-2016",
          },
          status: "GREEN",
          messages: [],
        },
      ],
    },
    whereYouLived: {
      residenceInfo: [
        {
          line1: "Addresss line 1",
          town: "Town",
          state: "State",
          postCode: "UL11223",
          countryCode: "UK",
          residenceType: "RENTING_FROM_AGENT",
          livedSinceMonth: 1,
          livedSinceYear: 2016,
          livedToMonth: null,
          livedToYear: null,
          reference: {
            contactFirstName: "Jonathan",
            contactLastName: "Beaumont",
            contactEmailAddress: "jonathan@gmail.com",
            referredAt: "01-01-2018",
          },
          status: "GREEN",
          messages: [],
        },
      ],
    },
  },
};
