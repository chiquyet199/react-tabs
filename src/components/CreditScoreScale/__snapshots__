/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/CreditScoreScale/__native_snapshots__