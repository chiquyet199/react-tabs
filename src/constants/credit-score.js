// THIS CODE IS COPIED FROM OLD "canopy-legacy" REPO
// Definition of common errors.
const NO_TRACE = "NO_TRACE";
const NO_TRACE_VALUE = -998;
const UNDEFINED_SCORE = "UNDEFINED_SCORE";
export const SCORE_MAX_VALUE = 999;
// Definition for the scores.
// Based on https://www.experian.co.uk/consumer/experian-credit-score.html
export const SCORES = {
  VERY_POOR: "VERY_POOR",
  POOR: "POOR",
  FAIR: "FAIR",
  GOOD: "GOOD",
  EXCELLENT: "EXCELLENT",
};

// Definition of the scores level within their ranges.
export const SCORE_RANGES = {
  [SCORES.VERY_POOR]: {from: 0, to: 560},
  [SCORES.POOR]: {from: 561, to: 720},
  [SCORES.FAIR]: {from: 721, to: 880},
  [SCORES.GOOD]: {from: 881, to: 960},
  [SCORES.EXCELLENT]: {from: 961, to: SCORE_MAX_VALUE},
};

// Values that are consider trusty.
const TRUSTY_SCORE_RANGE = [
  SCORES.POOR,
  SCORES.FAIR,
  SCORES.GOOD,
  SCORES.EXCELLENT,
];

/**
 * Get the score range based on the score literal.
 * @param  {String} score
 * @return {Object}
 */
export const getScoreRanges = score => SCORE_RANGES[score];

/**
 * A score shouldn't go beyond 999, which is the maximum at 'excellent' score level.
 * If the score goes beyond 999, just return 999.
 * @param  {Number} score
 * @return {Number}
 */
const normalizeScore = score =>
  Math.min(score, getScoreRanges(SCORES.EXCELLENT).to);

/**
 * Helper function to check if a score appears under a certain category.
 * @param  {Number} score
 * @param  {Object} range
 * @return {Boolean}
 */
const scoreIsInRange = (score, range) =>
  score >= range.from && score <= range.to;

/**
 * Check if the score is valued as 'no trace'.
 * @param  {Number}  score
 * @return {Boolean}
 */
const isNoTrace = score => score === NO_TRACE_VALUE;

/**
 * Check if the score is cataloged as 'very poor'.
 * @param  {Number}  score
 * @return {Boolean}
 */
const isVeryPoor = score =>
  scoreIsInRange(score, getScoreRanges(SCORES.VERY_POOR));

/**
 * Check if the score is cataloged as 'poor'.
 * @param  {Number}  score
 * @return {Boolean}
 */
const isPoor = score => scoreIsInRange(score, getScoreRanges(SCORES.POOR));

/**
 * Check if the score is cataloged as 'fair'.
 * @param  {Number}  score
 * @return {Boolean}
 */
const isFair = score => scoreIsInRange(score, getScoreRanges(SCORES.FAIR));

/**
 * Check if the score is cataloged as 'good'.
 * @param  {Number}  score
 * @return {Boolean}
 */
const isGood = score => scoreIsInRange(score, getScoreRanges(SCORES.GOOD));

/**
 * Check if the score is cataloged as 'excellent'
 * @param  {Number}  score
 * @return {Boolean}
 */
const isExcellent = score =>
  scoreIsInRange(score, getScoreRanges(SCORES.EXCELLENT));

/**
 * Get the score level (like 'VERY_POOR', 'POOR', etc.) for the score.
 * It will also check for possible errors like 'NO_TRACE'.
 *
 * @param  {Number} score
 * @return {String}
 */
export const getScoreLevel = score => {
  // Normalize the score, making it go as much as 999.
  const normalizedScore = normalizeScore(score);

  if (isNoTrace(normalizeScore)) {
    return NO_TRACE;
  }

  if (isVeryPoor(normalizedScore)) {
    return SCORES.VERY_POOR;
  }

  if (isPoor(normalizedScore)) {
    return SCORES.POOR;
  }

  if (isFair(normalizedScore)) {
    return SCORES.FAIR;
  }

  if (isGood(normalizedScore)) {
    return SCORES.GOOD;
  }

  if (isExcellent(normalizedScore)) {
    return SCORES.EXCELLENT;
  }

  return UNDEFINED_SCORE;
};

/**
 * Check if the score is consider as trusty.
 * @param  {Number}  score
 * @return {Boolean}
 */
export const isTrustyScore = score =>
  TRUSTY_SCORE_RANGE.includes(getScoreLevel(score));
