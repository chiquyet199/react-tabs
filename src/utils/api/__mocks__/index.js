import {injectMocks, extractScenarioFromLocation} from "data-mocks";
import example from "./test-mocks";
import users from "./users";

let scenario;

const {NODE_ENV} = process.env;

if (NODE_ENV !== "test") {
  scenario = extractScenarioFromLocation(window.location);
}

injectMocks(
  {
    default: [...example, ...users.default],
    failedSignup: [...users.failedSignup],
    takenEmail: [...users.takenEmail],
    failedToSendResetToken: [...users.failedToSendResetToken],
  },
  scenario,
);
