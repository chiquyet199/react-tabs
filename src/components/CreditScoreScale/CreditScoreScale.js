import React, {Component} from "react";
import styled from "styled-components/native";
import {colors} from "../../constants/theme";
import {PageTitle} from "../../components/common/layout";
import {getScoreLevel, SCORE_MAX_VALUE} from "../../constants/credit-score";

export const TextContainer = styled.View`
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
`;

export const ContentContainer = styled.View`
  max-width: 300px;
  width: 100%;
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: center;
  align-items: stretch;
  align-self: center;
  position: relative;
`;

export const ScaleImage = styled.Image`
  width: 220px;
  height: 178px;
`;

export const Heading = styled(PageTitle)`
  font-size: 20px;
  line-height: 20px;
  color: ${colors.dark};
  font-weight: 300;
  margin-bottom: 5px;
`;

export const LargeText = styled(PageTitle)`
  font-size: 55px;
  line-height: 55px;
  color: ${colors.dark};
`;

export const SmallText = styled.Text`
  font-size: 9px;
  line-height: 10px;
  color: ${colors.dark};
  padding-top: 3px;
  padding-horizontal: 15px;
  text-align: ${({left}) => (left ? "left" : "center")};
`;

export default class CreditScoreScale extends Component {
  getScaleImg = scoreLevel => {
    let scaleImg;
    switch (scoreLevel) {
      case "POOR":
        scaleImg = require("../../assets/rent_passport/financial/POOR.png");
        break;
      case "GOOD":
        scaleImg = require("../../assets/rent_passport/financial/GOOD.png");
        break;
      case "FAIR":
        scaleImg = require("../../assets/rent_passport/financial/FAIR.png");
        break;
      case "EXCELLENT":
        scaleImg = require("../../assets/rent_passport/financial/EXCELLENT.png");
        break;
      default:
        scaleImg = require("../../assets/rent_passport/financial/VERY_POOR.png");
    }

    return scaleImg;
  };

  renderScale = () => {
    const {translations, creditScore} = this.props;

    return (
      <TextContainer>
        <Heading>{translations.credit_score_label}</Heading>
        <LargeText>{creditScore}</LargeText>
        <SmallText>{`${
          translations.credit_score_out_of
        } ${SCORE_MAX_VALUE}`}</SmallText>
      </TextContainer>
    );
  };

  render() {
    const {creditScore, scaleImages} = this.props;
    let scaleImg;
    if (Object.keys(scaleImages).length) {
      scaleImg = scaleImages[getScoreLevel(creditScore)];
    } else {
      scaleImg = this.getScaleImg(getScoreLevel(creditScore));
    }

    return (
      <ContentContainer>
        <ScaleImage source={scaleImg} />
        {this.renderScale()}
      </ContentContainer>
    );
  }
}
