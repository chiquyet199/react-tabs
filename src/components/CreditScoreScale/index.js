import {connect} from "react-redux";
import CreditScoreScale from "./CreditScoreScale";

const mapStateToProps = (state, props) => ({
  creditScore: props.value || 0,
  translations: {
    credit_score_label: state.locale.translations.common.credit_score_label,
    credit_score_out_of: state.locale.translations.common.credit_score_out_of,
  },
});

export default connect(mapStateToProps)(CreditScoreScale);
