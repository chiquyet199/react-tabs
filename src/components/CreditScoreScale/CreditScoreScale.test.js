import React from "react";
import {shallow} from "enzyme";
import {initialState} from "../../__mocks__/initialState";
import configureStore from "redux-mock-store";
import CreditScoreScale, {
  TextContainer,
  ContentContainer,
  ScaleImage,
  Heading,
  LargeText,
  SmallText,
} from "./CreditScoreScale";
import locale from "../../utils/locales/en-US";
import {SCORE_MAX_VALUE} from "../../constants/credit-score";

const testImgPath = "FAIR.png";

const mockProps = {
  scaleImages: {},
  creditScore: 800,
  translations: {
    credit_score_label: locale.credit_score_label,
    credit_score_out_of: locale.credit_score_out_of,
  },
};

const mockStore = configureStore(mockProps);
describe("CreditScoreScale", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CreditScoreScale {...mockProps} />, {
      context: {store: mockStore(initialState)},
    });
  });

  it("renders as expected", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders properly TextContainer", () => {
    expect(wrapper.find(TextContainer)).toHaveLength(1);
  });
  it("renders properly ContentContainer", () => {
    expect(wrapper.find(ContentContainer)).toHaveLength(1);
  });

  it("renders properly ScaleImage", () => {
    expect(wrapper.find(ScaleImage)).toHaveLength(1);
  });

  it("renders properly Heading", () => {
    expect(wrapper.find(Heading)).toHaveLength(1);
    expect(wrapper.find(Heading).props().children).toEqual(
      mockProps.translations.credit_score_label,
    );
  });

  it("renders properly LargeText", () => {
    expect(wrapper.find(LargeText)).toHaveLength(1);
    expect(wrapper.find(LargeText).props().children).toEqual(
      mockProps.creditScore,
    );
  });

  it("renders properly SmallText", () => {
    expect(wrapper.find(SmallText)).toHaveLength(1);
    expect(wrapper.find(SmallText).props().children).toEqual(
      `${mockProps.translations.credit_score_out_of} ${SCORE_MAX_VALUE}`,
    );
  });

  it("renders FAIR Image according to given credit score", () => {
    expect(wrapper.find(ScaleImage).prop("source")).toEqual(testImgPath);
  });
});
