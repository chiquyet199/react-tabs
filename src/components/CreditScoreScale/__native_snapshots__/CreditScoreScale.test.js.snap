// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditScoreScale renders as expected 1`] = `
<Styled(Component)>
  <Styled(Image)
    source="FAIR.png"
  />
  <Styled(Component)>
    <Styled(Connect(WithTheme(Styled(Text))))>
      Credit Score
    </Styled(Connect(WithTheme(Styled(Text))))>
    <Styled(Connect(WithTheme(Styled(Text))))>
      800
    </Styled(Connect(WithTheme(Styled(Text))))>
    <Styled(Text)>
      out of 999
    </Styled(Text)>
  </Styled(Component)>
</Styled(Component)>
`;
